torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3326,7 +3326,9 @@ std::tuple<Tensor, Tensor> atan2_backward(
   if (!grad.defined()) {
     return std::tuple<Tensor, Tensor>{Tensor(), Tensor()};
   }
-  auto recip = (self * self + other * other).reciprocal();
+  auto denom = self * self + other * other;
+  auto recip = denom.reciprocal();
+  recip = at::where(denom == 0, at::zeros_like(recip), recip);
```

Comment:
In this case, I think the 2nd arg can be a scalar or scalar Tensor. No need to allocate a large zeros matrix.

Also wondering there is a way to make this where update inplace. Isn't it the same as 'recip[denom == 0] = 0' in python shorthand. Should be a way to do a selective assignment in CPP explicitly if the shorthand doesn't work.